Source/Core/Common/StringUtil.h

```diff
@@ -22,17 +23,11 @@
 namespace detail
 {
 template <typename T>
-constexpr bool IsBooleanEnum()
-{
-  if constexpr (std::is_enum_v<T>)
-  {
-    return std::is_same_v<std::underlying_type_t<T>, bool>;
-  }
-  else
-  {
-    return false;
-  }
-}
+using IsBooleanEnum =
+    std::is_same<bool, typename std::conditional_t<std::is_enum_v<T>, std::underlying_type<T>,
+                                                   std::type_identity<void>>::type>;
+template <typename T>
+concept BooleanEnum = IsBooleanEnum<T>::value;
```

Comment:
Can you leave the existing function and do 
```cpp
template <typename T>
concept BooleanEnum = IsBooleanEnum<T>();
```
The new one is much more confusing

Reply:
I suppose.

Reply:
Looks good